### DISCUSSION TITLE
Why use old Asp.net way on GPSTracker

### DISCUSSION THREAD

**Discussion by behroozbc** [dotnet/orleans#7867] [Q&A] [answered]
I see this [line of code](https://github.com/dotnet/orleans/blob/d700bb3eed98e16d4ea515f297c29417238ae27c/samples/GPSTracker/GPSTracker.Service/Program.cs#L5) that uses the old aspnet way.
https://github.com/dotnet/orleans/blob/d700bb3eed98e16d4ea515f297c29417238ae27c/samples/GPSTracker/GPSTracker.Service/Program.cs#L5
Why should we use this way instate of using the below approach? And what benefits the old method?
```

var builder = WebApplication.CreateBuilder(args);

builder.Host.UseOrleans((ctx, siloBuilder) => {

    siloBuilder.UseLocalhostClustering(
        siloPort: 11111 + instanceId,
        gatewayPort: 30000 + instanceId,
        primarySiloEndpoint: new IPEndPoint(IPAddress.Loopback, 11111));

});
.
.
.
var app = builder.Build();
...
...
...
app.Run();

```

**Comment by ReubenBond** [comment, marked as answer]
Either way is fine. The sample predates the Minimal API builder, as far as I recall. If you'd like, please submit a PR to update it to use the `WebApplicationBuilder`.

**Reply by behroozbc** [reply]
Thanks. It is my pleasure to help the Orleans team, and thanks a lot for the excellent video describing Orleans on dotnet shows.
